src/otx/algorithms/classification/adapters/mmcls/models/classifiers/custom_image_classifier.py

```diff
@@ -293,6 +294,21 @@ def extract_feat(self, img):
 
         return x
 
+    @custom_auto_fp16(apply_to=("img",))
+    def forward(self, img, return_loss=True, **kwargs):
```

Comment:
Do we need to cover forward methods of all the models we use by this decorator?
Does mixed-precision training work for other OTX models now?

Reply:
Can we avoid implicit modification of every model by applying the decorator in XPUDataParallel instead of autocast?

Reply:
> Can we avoid implicit modification of every model by applying the decorator in XPUDataParallel instead of autocast?

It seems impossible. Because we are using super call in XPUDataParallel, original decorator for `forward` must be called, and it is not desirable situation.
https://stackoverflow.com/questions/34206620/how-do-i-override-a-decorated-method-in-python

What do you think @eunwoosh? If I'm wrong, please correct it.

Reply:
Then, how about replacing `mmcv.runner.auto_fp16` to our `custom_auto_fp16`? To do that, replacing code should be executed before `import mmXX`. Similar to `import habana_frameworks.torch.gpu_migration` case, maybe we can place it at "src/otx/algorithms/common/utils/\_\_init\_\_.py".

Reply:
I checked it works well with https://github.com/openvinotoolkit/training_extensions/pull/2668/commits/f5cb459a79fb4340d39e5411b884f5ed1366241e.
![image](https://github.com/openvinotoolkit/training_extensions/assets/97221135/b72b7f56-f9e4-4f6e-9485-94a11cfb5eaf)